### PR TITLE
Support for lossless audio (flac) in hls

### LINKED ISSUE #1161: Support for Flac in HLS
Dear @erankor ,

Apple announced updates to the HLS spec at WWDC which make it possible to include FLAC lossless audio streams:

https://developer.apple.com/videos/play/wwdc2020/10225/
https://developer.apple.com/videos/play/wwdc2020/10158/

It would be great if nginx-vod-module could support this. 

https://apps.digitalconcerthall.com/temp/testfiles_flac_hls.zip (170 Mb) contains two files that could be used for testing:

- mp4 file with h264 video
- flac file with corresponding lossless audio

https://apps.digitalconcerthall.com.s3.amazonaws.com/temp/testfiles_flac_hls.zip
We are happy to test and / or provide further test files, of course. 

best regards
Marcus

### PR DESCRIPTION
**Pull request by jotair**
As apple now supports lossless audio (flac) in hls this pull adds this support mentioned in these issues

#1192 
Fixes #1161

We tested these changes in VLC and iOS / tvOS 14 devices with the default avFramework / avPlayer

### CODE REVIEW

**Comment by kaltura-hooks** [comment]
Hi @jotair,
Thank you for contributing this pull request!
Please sign the Kaltura CLA so we can review and merge your contribution.
Learn more at http://bit.ly/KalturaContrib

File: vod/codec_config.c [hunk 1]
```diff
@@ -500,7 +500,9 @@ codec_config_get_audio_codec_name(request_context_t* request_context, media_info
 	case VOD_CODEC_ID_EAC3:
 		codec_config_copy_string(media_info->codec_name, "ec-3");
 		return VOD_OK;
-
+	case VOD_CODEC_ID_FLAC:
+		codec_config_copy_string(media_info->codec_name, "fLaC");
+		return VOD_OK;
 	default:
```
Comment (erankor) [inline vod/codec_config.c:506]: please retain the newline separate between cases/default

File: vod/media_format.h [hunk 2]
```diff
@@ -126,7 +126,7 @@ enum {
 	VOD_CODEC_ID_VORBIS,
 	VOD_CODEC_ID_OPUS,
 	VOD_CODEC_ID_VOLUME_MAP,
-
+	VOD_CODEC_ID_FLAC,
 	// captions
```
Comment (erankor) [inline vod/media_format.h:130]: again, please keep the newline separator between the different types of codes

File: vod/mp4/mp4_parser.c [hunk 3]
```diff
@@ -2743,6 +2743,11 @@ mp4_parser_process_moov_atom_callback(void* ctx, atom_info_t* atom_info)
 			extra_data_required = FALSE;
 			break;
 
+		case FORMAT_FLAC:
+		      	metadata_parse_context.media_info.codec_id = VOD_CODEC_ID_FLAC;
+                        extra_data_required = FALSE;
+                        break;
```
Comment (erankor) [inline vod/mp4/mp4_parser.c:2749]: please fix the indent, need to use tabs here

**Review by erankor — CHANGES_REQUESTED** [review]

**Comment by erankor** [comment]
merged, thanks!

**Pull request merged**